### PR TITLE
Benchmark parametrisation

### PR DESCRIPTION
**Pull request by DurieuxPol**
Added the possibility to specify a directory other than /benchs to run the benchmarks from, in a similar way to how the number of iterations can be changed.
Also includes a workflow and a /test to run the test, as well as a bit of refactoring of how benchmarks are run.

### CODE REVIEW

**Comment by jordanmontt** [comment]
The Ci is failing with: `tests/benchParamTest.sh: not found`

**Comment by guillep** [comment]
> The Ci is failing with: `tests/benchParamTest.sh: not found`

This is probably a bug in the path used in the github action.

File: .github/workflows/benchparam.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: bench param test
+      uses: sudo-bot/action-shunit2@latest
+      with:
+        cli: "tests/benchParamTest.sh"
```
Comment (guillep) [inline .github/workflows/benchparam.yml:19]:
```suggestion
        cli: "./test/benchParamTest.sh"
```

Comment (guillep) [reply .github/workflows/benchparam.yml:19]: There is a typo in the directory name

**Review by guillep — CHANGES_REQUESTED** [review]

File: .github/workflows/benchparam.yml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  build:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: bench param test
+      uses: sudo-bot/action-shunit2@latest
+      with:
+        cli: "./test/benchParamTest.sh"
```
Comment (guillep) [inline .github/workflows/benchparam.yml:19]:
```suggestion
        cli: "bash ./test/benchParamTest.sh"
```

**Review by guillep — CHANGES_REQUESTED** [review]
the sh file has no execution permissions

**Comment by jordanmontt** [comment]
Nice, now we have an actual test failing

**Comment by guillep** [comment]
Tests are green! The smoke tests timeout because the druid build is super slow...

**Comment by guillep** [comment]
Thanks @DurieuxPol !

**Pull request merged**